interfaces/currency_rates_provider.py

```diff
@@ -0,0 +1,9 @@
+from abc import ABC, abstractmethod
+from typing import Union, Self
+
+class CurrencyRatesProvider(ABC):
+    """Returns dict with currency code as keys and CurrencyRate instances as values"""
+    @classmethod
+    @abstractmethod
+    def get_rates(cls: Self, currency_code: Union[str, None] = None) -> dict:
```

Comment:
I think another method is needed to handle getting rates for the specific currency

Reply:
Changed to return a tuple of CurencyRates